docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+classes will be accessed. If data is found on the NurseSchedules text
+file, it will be loaded into the arraylist, else a new text file is
+created.
+
+### Implementation:
+
+1.  User executes a command
+
+2.  NurseScheduleInstance calls UI.abortEnabledScanInput() to receive user input.
+
+3.  NurseScheduleInstance calls NurseScheduleParser.getFirstWord() to parse user input for specific commands.
+
+4.  Depending on the command, NurseScheduleInstance will call the relevant methods in NurseScheduleActions.
+
+5.  NurseScheduleActions will either add, list or delete a NurseSchedule object.
+
+### **Adding a new Nurse Schedule**
+
+### Implementation
+
+When the user attempts to add a new nurse schedule, the
+NurseScheduleStorage, NurseScheduleActions, UI and NurseScheduleUI
+classes will be assessed, and the following sequence of actions is
+called to prompt execution results to user:
+
+*add* \[Nurse ID\] \[Date (DDMMYYYY)\]:
+
+Getting User Input:
+
+1. User inputs add command which is processed by NurseScheduleInstance.runCommandLoopUntilExit().
+2. This calls NurseScheduleActions.addSchedule() which calls NurseScheduleUI.inputToCreateSchedule().
+
+Creating NurseSchedule object with User Input:
+
+3. NurseScheduleUI.inputToCreateSchedule() creates a new NurseSchedule object and is stored into an existing ArrayList\<NurseSchedule\>
+nurseSchedules which contains all the nurse schedule objects
+
+Saving NurseSchedule objects into .txt file:
+
+4. The command loop then calls NurseScheduleStorage.writeToFile() which starts the process of writing details of all existing Nurse Schedule objects within the ArrayList to the specified .txt file
+
+### **Listing Nurse Schedules**
+
+### Implementation:
+
+When the user attempts to list nurse schedules, they will have the
+choice of listing all schedules or a specified nurse id's schedule. This
+is similar to a search function. This will access the
+NurseScheduleActions class.
+
+*list* \[Nurse ID\] or list \[all\]:
+
+Getting User Input
+
+1. User inputs list \[Nurse ID/all\] command.
+2. Command loop calls NurseScheduleActions.listSchedules().
+
+Gathering necessary schedules
+
+3. listSchedules will call listAllSchedules() if the user inputs all, else it will check if Nurse ID is valid and call getNurseSchedulesById().
+
+Printing schedules
+
+4. printSchedules() is then called to print all schedules.
+
+### **Deleting Nurse Schedules**
+
+### Implementation:
+
+When the user wants to delete a specified nurse schedule, the NurseScheduleActions and NurseScheduleStorage classes will be accessed.
+
+*delete* \[Nurse ID\] \[Date (DDMMYYYY)\]:
+
+Getting User Input
+
+1. User inputs delete \[Nurse ID\] \[Date (DDMMYYYY)\] command
+2. Command loop calls NurseSchedulesActions.deleteSchedule().
+
+Deleting Schedule
+
+3. deleteSchedule() loops through the arraylist of schedules and calls remove() to delete the specified schedule. 
+
+### **Doctor Appointment-related Features**
+
+[Implementation:]{.ul}
```

Comment:
There is a few instances of such notations, "[description:]{.ul}". Does the notation have any significance? It would be good to put mention what the notation means in the glossary!